### PR TITLE
Fixes #1569 kill ice adapters on update

### LINKED ISSUE #1569: Update installation fails if ICE adapter is still open
**Describe the bug**
Sometimes the ICE adapter is still running after the client has been closed.
In that case the installation throws an error message that it can not create a file.

**To Reproduce**
Try to keep the ICE adapter alive? - then reinstall

**Expected behavior**
Installation works.

**Screenshots**
image

**OS**
Windows

Idea for solution: Maybe the install4j installer can check that no ICE adapter process is running before installation.

Reported by visionik.

**Wanna have the bug fixed quickly?**
Visit Issue hunt...
[Issue hunt](https://issuehunt.io/r/FAForever/downlords-faf-client)

### PR DESCRIPTION
**Pull request by 1-alex98**
Need to test this tho

### CODE REVIEW

File: downlords-faf-client.install4j [hunk 1]
```diff
@@ -699,6 +699,30 @@ return console.askOkCancel(message, true);
...
+              <action name="Kill left over ice adapters" id="621" customizedId="" beanClass="com.install4j.runtime.beans.actions.control.RunScriptAction" enabled="true" commentSet="false" comment="" actionElevationType="elevated" rollbackBarrier="false" rollbackBarrierExitCode="0" multiExec="false" failureStrategy="1" errorMessage="">
+                <serializedBean>
+                  <java class="java.beans.XMLDecoder">
+                    <object class="com.install4j.runtime.beans.actions.control.RunScriptAction">
+                      <void property="script">
+                        <object class="com.install4j.api.beans.ScriptProperty">
+                          <void property="value">
+                            <string> try{            
+     Runtime rt = Runtime.getRuntime();
+     Process proc = rt.exec("for /f \"tokens=1\" %i in ('jps -m ^| find \"faf-ice-adapter\"') do ( taskkill /F /PID %i )");
```
Comment (Geosearchef) [inline downlords-faf-client.install4j:741]: I don't like that, but I don't think there is an alternative.

**Review by Geosearchef — DISMISSED** [review]

File: downlords-faf-client.install4j (hunk 1 above)
Comment (1-alex98) [reply downlords-faf-client.install4j:741]: Well I suspect it won't hurt right. Also I need to test that first to

**Review by Geosearchef — DISMISSED** [review]
Why are there two?

**Comment by 1-alex98** [comment]
uninstaller and installer

**Comment by Geosearchef** [comment]
if this were normal java code I'd say replace the hardcoded `4000` in the sleep function with `Duration.ofSeconds(4).toMillis()` but as we have no imports here and would have to use the fully qualified name I'd keep it that way.

**Comment by 1-alex98** [comment]
@Geosearchef so approve it?

**Review by Geosearchef — APPROVED** [review]
I had my finger hovering over the approve button yesterday, but I didn't feel qualified enough to approve such a "hacky"/controversial change. 😅

**Comment by 1-alex98** [comment]
Well I built the installer and it works. That's what counts right. Dont like the idea either but it works so fuck it XD Would prefer the ice adpater to close better. It should exit if there is no connection to game for like a few minutes I guess.

**Pull request merged**